discovery/consul/consul.go

```diff
@@ -560,6 +583,30 @@ func (srv *consulService) watch(ctx context.Context, ch chan<- []*targetgroup.Gr
 			labels[taggedAddressesLabel+model.LabelName(name)] = model.LabelValue(v)
 		}
 
+		if serviceNode.Service.Kind != consul.ServiceKindTypical { // to avoid setting the label to an empty value
+			labels[serviceKindLabel] = model.LabelValue(serviceNode.Service.Kind)
+		}
+
+		if serviceNode.Service.Kind == consul.ServiceKindConnectProxy && serviceNode.Service.Proxy != nil {
+			labels[proxiedServiceLabel] = model.LabelValue(serviceNode.Service.Proxy.DestinationServiceName)
+			labels[proxiedServiceIDLabel] = model.LabelValue(serviceNode.Service.Proxy.DestinationServiceID)
+			labels[proxiedServiceAddressLabel] = model.LabelValue(serviceNode.Service.Proxy.LocalServiceAddress)
+			labels[proxiedServicePortLabel] = model.LabelValue(strconv.Itoa(serviceNode.Service.Proxy.LocalServicePort))
+
+			if srv.discovery.fetchProxyPaths {
+				for _, path := range serviceNode.Service.Proxy.Expose.Paths {
+					pathLabels := labels.Clone()
+					newAddr := strings.Replace(string(pathLabels[model.AddressLabel]), strconv.Itoa(serviceNode.Service.Port), strconv.Itoa(path.ListenerPort), 1)
```

Comment:
We should re-construct the label from scratch instead of using replace.

Reply:
ok!